[Ng E En Casper] iP

casperngeen opened the pull request:
# GFMD Demo

* Item 1

1. **This text is bold**
2. _This text is also italic_
3. ~~This text is strikethrough~~

```python 
def hello_world(): 
    print("Hello, World!") 
```

- [ ] Task 1
- [x] Task 2
- [ ] Task 3 :smile:

> This is [OpenAI](https://www.openai.com) and here is some `inline code`.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

casperngeen committed:
Renamed bot and added basic greet/exit

casperngeen committed:
Added echo and exit on bye command

casperngeen committed:
Added storage and retrieval of user input using the list command

casperngeen committed:
Created Task class

casperngeen committed:
Added commands to mark task as done or not done

casperngeen committed:
Created subclasses of Task

casperngeen committed:
Added tasktype field

casperngeen committed:
Add support for tracking todo, deadline and event task types

casperngeen committed:
Added files for I/O testing

casperngeen committed:
Updated input test & expected output

casperngeen committed:
Initalised exception package for CasperBotException

casperngeen committed:
Added exception handling

casperngeen committed:
Added delete feature

casperngeen committed:
Updated test files for deletion

casperngeen committed:
Made Task class abstract

casperngeen committed:
Add open/create file feature

casperngeen committed:
Add write and update file feature

casperngeen committed:
Update gitignore to ignore chatbot file

casperngeen committed:
Add date parsing and exception handling

casperngeen committed:
Intialise Parser class

casperngeen committed:
Initialise Storage class

casperngeen committed:
Initialise TaskList class

casperngeen committed:
Initialise Ui class

casperngeen committed:
Refactor to include more OOP

casperngeen committed:
Rename package to casper

casperngeen committed:
Merge branch 'add-gradle-support'

casperngeen committed:
Configure gradle

casperngeen committed:
Add JUnit testing

casperngeen committed:
Update error message

casperngeen committed:
Update error message UI

casperngeen committed:
Configure JAR build

casperngeen committed:
Add Javadoc

casperngeen committed:
Adjust imports to display all classes

casperngeen committed:
Add find feature

casperngeen committed:
Merge branch 'A-JavaDoc' to master

casperngeen committed:
Merge branch 'A-CodingStandard' to master

casperngeen committed:
Merge branch 'Level-9' to master

limyixiang commented on src/main/java/casper/Ui.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    public void greeting() {
+        System.out.println("Hello! I'm CasperBot.\n" +
```
Please place your line break before the "+" operator!

> In general:
> - Break after a comma.
> - Break before an operator. This also applies to the following "operator-like" symbols: the dot separator ., the ampersand in type bounds <T extends Foo & Bar>, and the pipe in catch blocks catch (FooException | BarException e)

limyixiang commented on src/main/java/casper/Ui.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    public void addTaskMessage(Task task) {
+        System.out.println("Got it. I've added this task:\n" +
```
"+" should be on a new line!

limyixiang commented on src/main/java/exception/CasperBotOutOfBoundsException.java:
```diff
@@ -0,0 +1,10 @@
+     * A custom exception for an invalid index input which is out of bounds
+     */
+    public CasperBotOutOfBoundsException() {
+        super("Index out of bounds", "Ensure that the index is within the length of the task list");
```
I think your line is too long! Please keep to the 120 characters limit!

limyixiang commented on src/main/java/casper/Ui.java:
```diff
@@ -0,0 +1,83 @@
+        System.out.println("------------------------------------------");
+    }
+
+    public void greeting() {
```
Would printGreeting be a better name for this method?

limyixiang commented on src/main/java/casper/CasperBot.java:
```diff
@@ -0,0 +1,199 @@
+     * @param commandType The type of command it should belong to (CREATE, TASK)
+     * @return
+     */
+    private static boolean isValidCommand(String command, CommandType commandType) {
```
I think method names should be in the form of a verb, `isValidCommand` is more suitable to be a boolean variable name.

limyixiang reviewed (commented):
Great work!

casperngeen committed:
Configure checkstyle

casperngeen committed:
Rewrite code in alignment with coding standard

casperngeen committed:
Rename file

luileng commented on src/test/java/casper/ParserTest.java:
```diff
@@ -0,0 +1,25 @@
+
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
```
Perhaps you could write header comments for public classes / methods such as this one

casperngeen committed:
Initialise the GUI using JavaFX

casperngeen committed:
Modify gradle to load GUI

casperngeen committed:
Reconfigure code to integrate with JavaFX and GUI

luileng commented on src/main/java/casper/Storage.java:
```diff
@@ -0,0 +1,154 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
```
I like how this follows coding standards

luileng commented on src/main/java/casper/CasperBot.java:
```diff
@@ -0,0 +1,209 @@
+import exception.CasperBotNumberFormatException;
+
+/**
+ * Represents the bot
```
Perhaps you can add a full stop for each header comment to follow coding standards

luileng commented on src/main/java/exception/CasperBotException.java:
```diff
@@ -0,0 +1,20 @@
+package exception;
```
I like the way you package your exceptions

luileng reviewed (commented):
The code generally adheres to coding standards, good job! However, there are some areas of improvement for the header comments.

casperngeen committed:
Configure FXML

casperngeen committed:
Merge branch 'branch-level-10'

casperngeen committed:
Add assertions

Some basic assertions were added, particularly at points where \n return values from other functions were used

casperngeen committed:
Refactor CasperBot to fit quality guidelines

Parts of the getResponse function were extracted into standalone functions, \n and I tried to use guards to handle exceptions, to ensure that the happy path was easily identifiable.

casperngeen committed:
Refactor Storage to improve code quality

casperngeen committed:
Refactor TaskList to improve code quality

casperngeen committed:
Merge pull request #1 from casperngeen/branch-A-Assertions

Add assertions

casperngeen committed:
Merge master to A-CodeQuality

casperngeen committed:
Merge pull request #2 from casperngeen/branch-A-CodeQuality

Merge code quality improvements

casperngeen committed:
Configure CI with Github Actions

casperngeen committed:
Add view schedule functionality

casperngeen committed:
Change bot display photo

casperngeen committed:
Update exception message for CasperBot

casperngeen committed:
Add user guide and UI

casperngeen committed:
Edit gradle to include package name

casperngeen committed:
Edit Ui.java to fix style issue

casperngeen committed:
Modify JavaDoc to fix typos

casperngeen committed:
Add title to GUI
